libraries/Networking/src/ESP_Network_Interface.cpp

```diff
@@ -0,0 +1,526 @@
+    }
+    char netif_name[8];
+    esp_err_t err = esp_netif_get_netif_impl_name(_esp_netif, netif_name);
+    if(err != ESP_OK){
+        log_e("Failed to get netif impl_name: %d", err);
+        return String("");
+    }
+    return String(netif_name);
+}
+
+uint8_t * ESP_Network_Interface::macAddress(uint8_t* mac)
+{
+    if(!mac || _esp_netif == NULL){
+        return NULL;
+    }
+    esp_err_t err = esp_netif_get_mac(_esp_netif, mac);
+    if(err != ESP_OK){
+        log_e("Failed to get netif mac: %d", err);
+        return NULL;
+    }
+    // getMac(mac);
+    return mac;
+}
+
+String ESP_Network_Interface::macAddress(void)
+{
+    uint8_t mac[6] = {0,0,0,0,0,0};
+    char macStr[18] = { 0 };
+    macAddress(mac);
+    sprintf(macStr, "%02X:%02X:%02X:%02X:%02X:%02X", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
+    return String(macStr);
+}
+
+IPAddress ESP_Network_Interface::localIP()
+{
+    if(_esp_netif == NULL){
+        return IPAddress();
+    }
+    esp_netif_ip_info_t ip;
+    if(esp_netif_get_ip_info(_esp_netif, &ip)){
+        return IPAddress();
+    }
+    return IPAddress(ip.ip.addr);
+}
+
+IPAddress ESP_Network_Interface::subnetMask()
+{
+    if(_esp_netif == NULL){
+        return IPAddress();
+    }
+    esp_netif_ip_info_t ip;
+    if(esp_netif_get_ip_info(_esp_netif, &ip)){
+        return IPAddress();
+    }
+    return IPAddress(ip.netmask.addr);
+}
+
+IPAddress ESP_Network_Interface::gatewayIP()
+{
+    if(_esp_netif == NULL){
+        return IPAddress();
+    }
+    esp_netif_ip_info_t ip;
+    if(esp_netif_get_ip_info(_esp_netif, &ip)){
+        return IPAddress();
+    }
+    return IPAddress(ip.gw.addr);
+}
+
+IPAddress ESP_Network_Interface::dnsIP(uint8_t dns_no)
+{
+    if(_esp_netif == NULL){
+        return IPAddress();
+    }
+    esp_netif_dns_info_t d;
+    if(esp_netif_get_dns_info(_esp_netif, dns_no?ESP_NETIF_DNS_BACKUP:ESP_NETIF_DNS_MAIN, &d) != ESP_OK){
+        return IPAddress();
+    }
+    return IPAddress(d.ip.u_addr.ip4.addr);
+}
+
+IPAddress ESP_Network_Interface::broadcastIP()
+{
+    if(_esp_netif == NULL){
+        return IPAddress();
+    }
+    esp_netif_ip_info_t ip;
+    if(esp_netif_get_ip_info(_esp_netif, &ip)){
+        return IPAddress();
+    }
+    return calculateBroadcast(IPAddress(ip.gw.addr), IPAddress(ip.netmask.addr));
+}
+
+IPAddress ESP_Network_Interface::networkID()
+{
+    if(_esp_netif == NULL){
+        return IPAddress();
+    }
+    esp_netif_ip_info_t ip;
+    if(esp_netif_get_ip_info(_esp_netif, &ip)){
+        return IPAddress();
+    }
+    return calculateNetworkID(IPAddress(ip.gw.addr), IPAddress(ip.netmask.addr));
+}
+
+uint8_t ESP_Network_Interface::subnetCIDR()
+{
+    if(_esp_netif == NULL){
+        return (uint8_t)0;
+    }
+    esp_netif_ip_info_t ip;
+    if(esp_netif_get_ip_info(_esp_netif, &ip)){
+        return (uint8_t)0;
+    }
+    return calculateSubnetCIDR(IPAddress(ip.netmask.addr));
+}
+
+IPAddress ESP_Network_Interface::localIPv6()
```

Comment:
This is not consistent with usage for IPv4. This is not the local (as opposed to remote) address, with global scope, but the link local scope address.

It should be called `linkScopeIPv6()` or `linkLocalIPv6()` or something similar.